### DISCUSSION TITLE
Multi-region cross-account SQS client selection

### DISCUSSION THREAD

**Discussion by ArranOakes** [awspring/spring-cloud-aws#884] [Q&A]
Hi there, I appreciate some of this would typically be achieved using IAM and Assumed Roles, but I have a use-case where I am using multiple AWS services (SQS, SNS and S3) across multiple accounts and across multiple regions. The permutations can get a little crazy.

With SQS in particular, I need to produce and consume from different regions cross-account, so I know that by defining my own SQS client bean I can override the autoconfiguration, but is it possible to, via a @Qualifier/Named Bean or similar approach, select which client a specific @SqsListener is using, so that my listeners can connect using different clients with different configurations?

Thanks!

**Comment by tomazfernandes** [comment]
Hi @ArranOakes-ELS.

You should be able to use the [SqsListenerContainerFactory](https://docs.awspring.io/spring-cloud-aws/docs/3.0.2/reference/html/index.html#sqsmessagelistenercontainerfactory) to set different `SqsAsyncClient` instances, and then refer to the appropriate factory in the `@SqsListener` annotation.

Would this work for you?

Thanks.
